src/Misc/layoutbin/update.sh.template

```diff
@@ -125,6 +125,14 @@ date "+[%F %T-%4N] Update succeed" >> "$logfile"
 date "+[%F %T-%4N] Rename $logfile to be $logfile.succeed" >> "$logfile" 2>&1
 mv -fv "$logfile" "$logfile.succeed" >> "$logfile" 2>&1
 
+# reinstall service if needed (update systemd/launchd template)
+if [[ -e "$rootfolder/.service" ]]
+then
+    "$rootfolder/svc.sh" uninstall
```

Comment:
this will be a problem for Linux runner since service in Linux is installed via [sudo ./svc.sh install](https://docs.github.com/en/free-pro-team@latest/actions/hosting-your-own-runners/configuring-the-self-hosted-runner-application-as-a-service#installing-the-service) during runner config time, the update script is running under runner execution context, might not have permission to `sudo`.

Reply:
The other "OS-agnostic" solution I can see here is to remove this part, change service settings to always restart watchdog and make watchdog exits on update. However this service update will still require user intervention on Linux.

We can also run the service reinstall exclusively on Darwin, but it will be, for now, the only UNIX OS to handle service template updates.